opensrp-reveal/src/main/java/org/smartregister/reveal/fragment/ChildRegisterFragment.java

```diff
@@ -0,0 +1,308 @@
+package org.smartregister.reveal.fragment;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.vijay.jsonwizard.domain.Form;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.smartregister.domain.FetchStatus;
+import org.smartregister.domain.ResponseErrorStatus;
+import org.smartregister.receiver.SyncStatusBroadcastReceiver;
+import org.smartregister.reveal.R;
+import org.smartregister.reveal.adapter.ChildRegisterAdapter;
+import org.smartregister.reveal.adapter.GroupedListableAdapter;
+import org.smartregister.reveal.contract.BaseDrawerContract;
+import org.smartregister.reveal.contract.ChildRegisterFragmentContract;
+import org.smartregister.reveal.contract.FormProcessor;
+import org.smartregister.reveal.model.Child;
+import org.smartregister.reveal.model.ChildModel;
+import org.smartregister.reveal.presenter.ChildRegisterFragmentPresenter;
+import org.smartregister.reveal.util.Constants;
+import org.smartregister.reveal.view.ChildProfileActivity;
+import org.smartregister.reveal.view.ChildRegisterActivity;
+import org.smartregister.reveal.view.DrawerMenuView;
+import org.smartregister.reveal.viewholder.GroupedListableViewHolder;
+import org.smartregister.util.Utils;
+import org.smartregister.view.fragment.BaseListFragment;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import timber.log.Timber;
+
+import static org.smartregister.reveal.util.Constants.JsonForm.ENCOUNTER_TYPE;
+
+public class ChildRegisterFragment extends BaseListFragment<Child> implements ChildRegisterFragmentContract.View, BaseDrawerContract.DrawerActivity, SyncStatusBroadcastReceiver.SyncStatusListener,
+        FormProcessor.Requester {
+    public static final String TAG = "ChildRegisterFragment";
+
+    private BaseDrawerContract.View drawerView;
+    private TextView tvTitle;
+    @Nullable
+    private HashMap<String, List<String>> filterAndSearch;
+    private String locationName = "";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'locationName' could be replaced by a local variable.](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5414138)